xline/src/storage/lease_store/mod.rs

```diff
@@ -155,14 +164,41 @@ where
     pub(crate) fn is_primary(&self) -> bool {
         self.is_primary.load(Ordering::Relaxed)
     }
+
+    /// Make lease synced, remove it from `unsynced_cache`
+    pub(crate) async fn mark_lease_synced(&self, wrapper: &RequestWrapper) {
+        #[allow(clippy::wildcard_enum_match_arm)] // only the following two type are allowed
+        let lease_id = match *wrapper {
+            RequestWrapper::LeaseGrantRequest(ref req) => req.id,
+            RequestWrapper::LeaseRevokeRequest(ref req) => req.id,
+            _ => {
+                return;
+            }
+        };
+
+        _ = self.unsynced_cache.write().await.remove(&lease_id);
+        self.sync_event.notify_waiters();
```

Comment:
The other parts of `Xline` use `eventer_listener` to implement similar abilities, which way is batter? should we unify them

Reply:
It seems the `event_listener` is synchronous, the `tokio::sync::Notify` will be better?

Reply:
It can also be used asynchronous

Reply:
I think I better use `event_listener` for better consistency.